File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/DTOs/BookDTO.java
package com.nokia.library.nokiainnovativeproject.DTOs;

import com.nokia.library.nokiainnovativeproject.entities.BookDetails;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;
import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class BookDTO {

	@Size(max = 100, message = "Signature can't exceed 100 characters")
	@NotBlank(message = "Signature is required")
	private String signature;

	@Size(max = 5000, message = "Comments can't exceed 5000 characters")
	private String comments;

	@NotNull(message = "Book details is required.")
	private Long bookDetailsId;

    @NotNull(message = "Book status is required.")
	private Long bookStatusId;
}

File: frontend/src/app/models/database/DTOs/BookDTO.ts
export class BookDTO {
	signature: string;
	bookDetailsId: number;
	comments: string;
	bookStatusId: number;
	owners: number[];

	constructor(signature: string, bookDetailsId: number, comments: string, bookStatusId: number, owners: number[]) {
		this.signature = signature;
		this.bookDetailsId = bookDetailsId;
		this.comments = comments;
		this.bookStatusId = bookStatusId;
		this.owners = owners;
	}
}